src/app/components/mnoe-api/admin/provisioning.svc.coffee

```diff
@@ -48,15 +48,14 @@
     subscription
 
   # Return the subscription
+  # if editing return the subscription in cache (edition mode)
   # if productNid: return the default subscription
   # if subscriptionId: return the fetched subscription
-  # else: return the subscription in cache (edition mode)
-  @initSubscription = ({productNid = null, subscriptionId = null, orgId = null}) ->
+  # else: return an empty object (this condition should never be met, but is a failsafe)
+  @initSubscription = ({productNid = null, subscriptionId = null, orgId = null, editing = false}) ->
```

Comment:
@ouranos Yeah, that was my first inclination too. 

The problem with that is for new subscriptions that do not yet have an id, those only have a productNid.

Reply:
As per the first comment, yes thats definitely better.

Reply:
I'm trying to understand how this fits into the picture, could it be something like:

```js
# When loading the Subscription form
@initSubscription = ({productNid = null, subscriptionId = null, orgId = null, editing = false}) ->
    deferred = $q.defer()

  # Subscription already loaded, reuse it
  if(subscription)
    deferred.resolve(subscription)
  else if subscriptionId?
    ...
  else if productNid?
    ...
  else
    deferred.resolve({})

    return deferred.promise

# Remove reference to subscription when closing workflow (save, cancel, etc...)
@saveSubscription = (subscription) ->
  ...
  subscription = null
```

Reply:
@BrunoChauvet That was one of my thoughts too -- the problem I ran into was this: 

Scenario (without refreshing).
1) You go through subscription workflow with Xero
2) You click out of subscription workflow (don't save). 
3) You go to subscription workflow with Quickbooks 
Result: Xero's information shows.

Now, this may not be a problem, as later on in OPAL-432, I made it so that when you either edit the subscription or create a new subscription, you reset the service's subscription to an empty object.

This then may be best solution.